tests/rados/test_stretch_osd_serviceability_scenarios.py

```diff
@@ -0,0 +1,332 @@
+    )
+    test_host = rados_obj.fetch_host_node(daemon_type="osd", daemon_id=target_osd)
+    should_not_be_empty(test_host, "Failed to fetch host details")
+    dev_path = get_device_path(test_host, target_osd)
+    log.debug(
+        f"osd device path  : {dev_path}, osd_id : {target_osd}, hostname : {test_host.hostname}"
+    )
+    utils.set_osd_devices_unmanaged(ceph_cluster, target_osd, unmanaged=True)
+    method_should_succeed(utils.set_osd_out, ceph_cluster, target_osd)
+    method_should_succeed(wait_for_clean_pg_sets, rados_obj, timeout=12000)
+    log.debug("Cluster clean post draining of OSD for removal")
+    utils.osd_remove(ceph_cluster, target_osd)
+    method_should_succeed(wait_for_clean_pg_sets, rados_obj, timeout=12000)
+    method_should_succeed(utils.zap_device, ceph_cluster, test_host.hostname, dev_path)
+    method_should_succeed(wait_for_device, test_host, target_osd, action="remove")
+
+    # Waiting for recovery to post OSD host removal
+    method_should_succeed(wait_for_clean_pg_sets, rados_obj, timeout=12000)
+
+    # Checking if the expected health warning about the different site weights are seen.
+    if not check_stretch_health_warning():
+        log.error("Warnings not generated on the cluster")
+        raise Exception("Warning not present on cluster")
+
+    # Checking cluster health after OSD removal
+    method_should_succeed(rados_obj.run_pool_sanity_check)
+    log.info(
+        f"Removal of OSD : {target_osd} is successful. Proceeding to add back the OSD daemon."
+    )
+    if not rados_obj.check_inactive_pgs_on_pool(pool_name=pool_name):
+        log.error(f"Inactive PGs found on pool : {pool_name}")
+        inactive_pgs += 1
+
+    log.debug(
+        f"Ceph osd tree after OSD removal : \n\n {rados_obj.run_ceph_command(cmd='ceph osd tree')} \n\n"
+    )
+
+    # Adding the removed OSD back and checking the cluster status
+    log.debug("Adding the removed OSD back and checking the cluster status")
+    utils.add_osd(ceph_cluster, test_host.hostname, dev_path, target_osd)
+    method_should_succeed(wait_for_device, test_host, target_osd, action="add")
+    time.sleep(30)
+    log.debug(
+        "Completed addition of OSD post removal. Checking for inactive PGs post OSD addition"
+    )
+    if not rados_obj.check_inactive_pgs_on_pool(pool_name=pool_name):
+        log.error(f"Inactive PGs found on pool : {pool_name}")
+        inactive_pgs += 1
+
+    if inactive_pgs > 10:
+        log.error("Found inactive PGs on the cluster during OSD removal/Addition")
+        raise Exception("Inactive PGs during stop error")
+
+    # Checking if the expected health warning about the different site weights is removed post addition of OSD.
+    if check_stretch_health_warning():
+        log.error("Warnings not removed on the cluster post osd addition")
+        raise Exception("Warning present on cluster")
+
+    # Checking cluster health after OSD removal
```

Comment:
minor comment in section - checking cluster health after OSD addition.

Reply:
Updated.